cpp/src/progress_thread.cpp

```diff
@@ -113,6 +117,7 @@ void ProgressThread::pause() {
 
 void ProgressThread::resume() {
     thread_.resume();
+    cv_.notify_all();
```

Comment:
We need to notify on state changes in both directions.

Reply:
Isn't this already achieved with https://github.com/rapidsai/rapidsmpf/pull/365/files#diff-07f80ad017d4088866533997c3a66643e01ef6bd52c0d53ef60dc69afa9d5a4dR79?

Reply:
No? This one notifies waiters in the progressthread that its internal pausablethread has been updated in state to running.